### PR TITLE
[CHORE] unit tests heuristic four

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[CHORE](https://bcdevex.atlassian.net/browse/CHORE)

Objective: 

- unit test round four

### CODE REVIEW

File: apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts [hunk 1]
```diff
@@ -410,4 +412,83 @@ describe('PosReconciliationService', () => {
       ).toBe(true);
     });
   });
+  describe('heuristic round four', () => {
+    beforeAll(() => {
+      // create some data that will not match
+      const data = new MockData(PaymentMethodClassification.POS);
+      payments = data.paymentsMock as PaymentEntity[];
+      deposits = data.depositsMock as POSDepositEntity[];
+      /*eslint-disable */
```
Comment (fw-noel) [inline apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts:421]: Why?

File: apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts [hunk 2]
```diff
@@ -410,4 +412,83 @@ describe('PosReconciliationService', () => {
...
+      roundFourDepositMatches = matchedRoundFour.flatMap((itm) => itm.deposits);
+    });
+    it('should aggregate the deposits and build the deposit dictionary', () => {
+      const aggregatedDeposits = service.aggregateDeposits(
+        deposits
+      ) as unknown as POSDepositEntity[];
+
+      const depositsDictionary =
+        service.buildPosDepositsDictionary(aggregatedDeposits);
+      console.log(depositsDictionary);
```
Comment (fw-noel) [inline apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts:450]: Log

File: apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts [hunk 3]
```diff
@@ -410,4 +412,83 @@ describe('PosReconciliationService', () => {
...
+
+      expect(dictKeys).toEqual(
+        expect.arrayContaining(
+          aggregatedDeposits.map((itm) => itm.transaction_amt.toString())
+        )
+      );
+    });
+
+    it('should match based on date, aggregated amount, method ', () => {
+      console.log(service.aggregatePayments(roundFourPaymentMatches));
```
Comment (fw-noel) [inline apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts:462]: Logs

File: apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts [hunk 4]
```diff
@@ -410,4 +412,83 @@ describe('PosReconciliationService', () => {
...
+      const reaggregatedPayemnts = service.aggregatePayments(
+        roundFourPaymentMatches
+      );
+      const reaggregatedDeposits = service.aggregateDeposits(
+        roundFourDepositMatches
+      );
+      expect(
+        reaggregatedPayemnts
+          .map(
+            (payment) =>
```
Comment (fw-noel) [inline apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts:473]: Might suggest being explicit and naming these inner variables `aggregatedPayment` and `aggregatedDeposit`. We want to be clear in avoiding confusion between payments and aggregated payments

File: apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts [hunk 5]
```diff
@@ -410,4 +412,83 @@ describe('PosReconciliationService', () => {
...
+      ).toBe(true);
+    });
+    it('should match many to many', () => {
+      const matchedDepositIds = roundFourDepositMatches.map((itm) => itm.id);
+      const matchedDepositsFromMatchedPayments =
+        roundFourPaymentMatches.flatMap((itm) =>
+          itm.round_four_matches?.map((itm) => itm.id)
+        );
+      expect(matchedDepositIds).toEqual(
+        expect.arrayContaining(matchedDepositsFromMatchedPayments)
```
Comment (fw-noel) [inline apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts:490]: `expect.arrayContaining` doesn't mean a 1:1 match, correct? Maybe calls for a more explicit test?

**Review by fw-noel — COMMENTED** [review]
Looks okay for the most part! Just a couple comments

File: apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts (hunk 5 above)
Comment (chelsea-EYDS) [reply apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts:490]: toEqual should be strict equality? I will verify...

**Review by fw-noel — APPROVED** [review]
Looks good, would love to see the last comment addressed

**Review by fw-noel — APPROVED** [review]
Looks good, would love to see the last comment addressed

**Pull request merged**